ambry-store/src/main/java/com.github.ambry.store/StoreMessageReadSet.java

```diff
@@ -146,6 +148,34 @@ public void close() {
       segment.closeView();
     }
   }
+
+  /**
+   * Do data preFetch: from disk to memory buffer.
+   * @param relativeOffset the relativeOffset to start.
+   * @param size The size requested to preFetch.
+   */
+  public void preFetch(long relativeOffset, long size) {
+    long sizeToRead = Math.min(size, getMessageInfo().getSize() - relativeOffset);
```

Comment:
feel like we should throw an exception if `size > getMessageInfo().getSize() - relativeOffset`

Reply:
fixed.